irods_specific_implementation/irods_rest_logical_path_delete_api_implementation.h

```diff
@@ -0,0 +1,88 @@
+#include "irods_rest_api_base.h"
+
+#include "constants.hpp"
+#include <irods/filesystem.hpp>
+#include <irods/rodsClient.h>
+#include <irods/irods_query.hpp>
+
+#include <pistache/http_headers.h>
+#include <pistache/optional.h>
+#include <pistache/router.h>
+
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+namespace irods::rest
+{
+    // this is contractually tied directly to the api implementation
+    const std::string service_name{"irods_rest_cpp_logical_path_delete_server"};
+
+    class logical_path_delete : public api_base
+    {
+    public:
+        logical_path_delete()
+            : api_base{service_name}
+        {
+            info("Endpoint initialized.");
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        operator()(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            namespace fs = irods::experimental::filesystem;
+
+            try {
+
+                auto _logical_path = _request.query().get("logical-path").get();
+                auto _no_trash     = _request.query().get("no-trash").getOrElse("0");
+                auto _verbose      = _request.query().get("verbose").getOrElse("0");
+                auto _progress     = _request.query().get("progress").getOrElse("0");
+                auto _recursive    = _request.query().get("recursive").getOrElse("0");
+                auto _unregister   = _request.query().get("unregister").getOrElse("0");
+
+                bool no_trash      = http_bool(_no_trash);
+                bool verbose       = http_bool(_verbose);
+                bool progress      = http_bool(_progress);
+                bool recursive     = http_bool(_recursive);
+                bool unregister    = http_bool(_unregister);
+
+                auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+                const fs::path logical_path ( _logical_path.c_str() );
+
+                fs::extended_remove_options opts {
+                    .no_trash      = no_trash,
+                    .verbose       = verbose,
+                    .progress      = progress,
+                    .recursive     = recursive,
+                    .unregister    = unregister
+                };
+
+                bool status  = fs::NAMESPACE_IMPL::remove(*conn(), logical_path, opts);
+
+                nlohmann::json json_obj = {
+                    {"op_status", status ? op_success : op_failure}
+                };
+                return std::make_tuple(
+                  !status ? Pistache::Http::Code::Bad_Request : Pistache::Http::Code::Ok,
+                  json_obj.dump()
+                );
+            }
+            catch (const fs::filesystem_error& e) {
+                error("Caught exception - [error_code={}] {}", e.code().value(), e.what());
+                return make_error_response(e.code().value(), e.what());
+            }
```

Comment:
How does this error case relate to the `json_obj` defined above?
How helpful is the use of `json_obj["op_status"]`?
Should this REST API operation return anything if an exception isn't thrown?

*These questions apply to irods_rest_cpp_logical_path_rename_server as well.*

Reply:
This troubled me too; I think it makes sense to change it to return nothing.